XenAdmin/Actions/GUIActions/ExternalPluginAction.cs

```diff
@@ -246,21 +246,21 @@ void _extAppProcess_OutputDataReceived(object sender, DataReceivedEventArgs e)
         }
 
         // Returns a set of params which relate to the object you have selected in the treeview
-        private List<string> RetrieveParams(IXenObject obj)
+        private IEnumerable<string> RetrieveParams(IXenObject obj)
         {
-            IXenConnection connection = obj.Connection;
-            Host coordinator = connection != null ? Helpers.GetCoordinator(connection) : null; // get coordinator asserts connection is not null
-            string coordinatorAddress = EmptyParameter;
+            var connection = obj.Connection;
+            var coordinator = connection != null ? Helpers.GetCoordinator(connection) : null; // get coordinator asserts connection is not null
+            var coordinatorAddress = EmptyParameter;
 
             if (coordinator != null)
             {
                 coordinatorAddress = Helpers.GetUrl(coordinator.Connection);
                 WriteTrustedCertificates(coordinator.Connection);
             }
 
-            string sessionRef = connection.Session != null ? connection.Session.opaque_ref : EmptyParameter;
-            string objCls = obj != null ? obj.GetType().Name : EmptyParameter;
-            string objUuid = obj != null && connection.Session != null ? Helpers.GetUuid(obj) : EmptyParameter;
+            var sessionRef = connection?.Session != null ? connection.Session.opaque_ref : EmptyParameter;
+            var objCls = obj.GetType().Name;

```

Comment:
I removed the null check because IXenObject isn't nullable.

Here's Visual Studio's take on it:
![image](https://user-images.githubusercontent.com/32554698/140038611-1807662c-ed86-4776-81b0-7e768e48f578.png)